extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-logs.js

```diff
@@ -221,4 +198,60 @@ export class AmpStoryDevToolsTabLogs extends AMP.BaseElement {
     }
     this.mutateElement(() => logsTabSelector.appendChild(statusIcon));
   }
+
+  /**
+   * @private
+   * @param {!Element} logsContainer
+   */
+  addSuccessMessageToLogs_(logsContainer) {
+    logsContainer.classList.add('i-amphtml-story-dev-tools-logs-success');
+    logsContainer.appendChild(
+      createElementWithAttributes(this.element.ownerDocument, 'div', {
+        'class': 'i-amphtml-story-dev-tools-logs-success-image',
+      })
+    );
+    const successMessage = createElementWithAttributes(
+      this.element.ownerDocument,
+      'div',
+      {
+        'class': 'i-amphtml-story-dev-tools-logs-success-message',
+      }
+    );
+    successMessage.textContent = `Great Job!\r\nNo issues found.`;
+    logsContainer.appendChild(successMessage);
+  }
+
+  /**
+   * @private
+   * @param {!Element} logsContainer
+   */
+  addErrorsToLogs_(logsContainer) {
+    this.errorList_.forEach((content) => {
+      const logEl = buildLogMessageTemplate(this.element);
+      logEl.querySelector('.i-amphtml-story-dev-tools-log-type').textContent =
+        content.code;
+      const codeEl = logEl.querySelector('.i-amphtml-story-dev-tools-log-code');
+      content.htmlLines.forEach((l, i) => {
+        const lineEl = this.element.ownerDocument.createElement('span');
+        lineEl.classList.add('i-amphtml-story-dev-tools-log-code-line');
+        lineEl.textContent = (i + content.line - 1).toString() + '|' + l;
+        codeEl.appendChild(lineEl);
+      });
+      logEl.querySelector(
+        '.i-amphtml-story-dev-tools-log-position'
+      ).textContent = `${content.line}:${content.col}`;
+      logEl.querySelector(
+        '.i-amphtml-story-dev-tools-log-description'
+      ).textContent = content.message;
+      const specUrlElement = logEl.querySelector(
+        '.i-amphtml-story-dev-tools-log-spec'
+      );
+      if (content.specUrl) {
+        specUrlElement.href = content.specUrl;
+      } else {
+        specUrlElement.remove();
+      }
+      logsContainer.appendChild(logEl);
+    });
+  }
```

Comment:
For both code clarity and performance, please refactor this code to follow the following structure:
1. Build the DOM
2. In a `mutateElement`, append the entire template at once

Reply:
That's what I'm doing right now (check the function calls, they create the content and then get attached in a mutate context), but probably the method names could be renamed or commented better